src/main/java/org/terasology/gooeyDefence/worldGeneration/rasterizers/DefenceFieldRasterizer.java

```diff
@@ -54,10 +54,10 @@ public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         for (Map.Entry<Vector3i, Boolean> entry : fieldFacet.getWorldEntries().entrySet()) {
             if (entry.getValue()) {
                 Vector3i pos = entry.getKey();
-                if ((int) DefenceField.distanceToNearestEntrance(JomlUtil.from(pos)) < DefenceField.entranceRingSize + 2) {
-                    chunk.setBlock(ChunkMath.calcRelativeBlockPos(pos), altBlock);
+                if ((int) DefenceField.distanceToNearestEntrance(pos) < DefenceField.entranceRingSize + 2) {
+                    chunk.setBlock(ChunkMath.calcRelativeBlockPos(pos, new Vector3i()), altBlock);
```

Comment:
Same remarks as for the Exoplanet PR: I think we can avoid additional vector allocations inside the loop by introducing a temporary vector outside the loop and reusing it for those computations.

I think we can see a best practice/bad smell pattern evolve here, which we should note down _somewhere_.